### PR TITLE
Change Nutanix credentials format

### PR DESCRIPTION
**Pull request by thunderboltsid**
Update Nutanix Secret template to use base64 encoded credentials.

### CODE REVIEW

**Comment by thunderboltsid** [comment]
/hold 
This is an anticipatory PR; requires a new version of CAPX before we merge this.

**Comment by thunderboltsid** [comment]
PR to change the CAPX version that supports new format https://github.com/aws/eks-anywhere-build-tooling/pull/1433

File: pkg/providers/nutanix/template.go
Comment (abhinavmpandey08) [inline pkg/providers/nutanix/template.go:93]: Doesn't look like we need the `credentials.NutanixCredentials` struct. We can use `credentials.Credentials` directly
```suggestion
	nutanixCreds := []credentials.Credential{{
		Type: credentials.BasicAuthCredentialType,
		Data: encodedCreds,
	}},
	credsJSON, _ := json.Marshal(nutanixCreds)
```

File: pkg/providers/nutanix/template.go
Comment (abhinavmpandey08) [inline pkg/providers/nutanix/template.go:86]: Let's not ignore the error here
```suggestion
	encodedCreds, err := json.Marshal(ntb.creds)
	if err != nil {
		return nil, fmt.Errorf("marshaling Nutanix credentials: %v", err)
	}
```
Same on line 93

**Review by abhinavmpandey08 — APPROVED** [review]
/approve

**Pull request merged**
